lib/msal-node/test/client/ManagedIdentitySources/AzureArc.spec.ts

```diff
@@ -110,6 +119,62 @@ describe("Acquires a token successfully via an Azure Arc Managed Identity", () =
             );
         });
 
+        test("acquires a token when both/either the identityEndpoint and/or imdsEndpoint environment variables are undefined, and the himds executable exists and its permissions allow it to be read", async () => {
+            // delete the environment variables so the himds executable is checked
+            delete process.env[
+                ManagedIdentityEnvironmentVariableNames.IDENTITY_ENDPOINT
+            ];
+            delete process.env[
+                ManagedIdentityEnvironmentVariableNames.IMDS_ENDPOINT
+            ];
+            // delete value cached from getManagedIdentitySource() in the beforeEach
+            delete ManagedIdentityClient["sourceName"];
+
+            // MI source will not be Azure Arc yet, since the environment variables are undefined,
+            // and accessSyncSpy still returns an error
+            // (meaning either the himds file doesn't exists or its permissions don't allow it to be read)
+            expect(
+                managedIdentityApplication.getManagedIdentitySource()
+            ).not.toBe(ManagedIdentitySourceNames.AZURE_ARC);
+            // delete value cached from getManagedIdentitySource() directly above
+            delete ManagedIdentityClient["sourceName"];
+
+            // returns undefined when the himds file exists and its permissions allow it to be read
+            // otherwise, throws an error
+            accessSyncSpy.mockImplementationOnce(() => {
```

Comment:
What does returning undefined do? I am not sure I understand this mock and its purpose. Sorry some JS language gaps. Can you please explain this?

Reply:
[NodeJS native `accessSync` function](https://nodejs.org/api/fs.html#fsaccesssyncpath-mode) (used to check if file exists and its permissions) returns undefined on truthy, and throws error on falsey.